app/scripts/controllers/transactions/index.js

```diff
@@ -357,6 +367,10 @@ class TransactionController extends EventEmitter {
       type: TRANSACTION_TYPE_RETRY,
     })
 
+    if (customGasLimit) {
+      newTxMeta.txParams.gas = customGasLimit
```

Comment:
Can we pass this in to `generateTxMeta` above?

Reply:
We could, but I thought this would be better. If I was to pass it into `generateTxMeta`, I'd have to create a fallback to handle the case where `customGasLimit` isn't passed in. This seemed more straightforward to me 🤷‍♂